File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.js

var MyQueue = function() {
    this.stack = [];
};

/** 
 * @param {number} x
 * @return {void}
 */
MyQueue.prototype.push = function(x) {
    const newStack = [];
    while (this.stack.length > 0) {
        newStack.push(this.stack.pop());
    }
    newStack.push(x);
    while (newStack.length > 0) {
        this.stack.push(newStack.pop());
    }  
};

/**
 * @return {number}
 */
MyQueue.prototype.pop = function() {
    return this.stack.length > 0 ? this.stack.pop() : undefined;
};

/**
 * @return {number}
 */
MyQueue.prototype.peek = function() {
    return this.stack.length > 0 ? this.stack[this.stack.length - 1] : undefined;
};

/**
 * @return {boolean}
 */
MyQueue.prototype.empty = function() {
    return this.stack.length === 0;
};

/** 
 * Your MyQueue object will be instantiated and called as such:
 * var obj = new MyQueue()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.peek()
 * var param_4 = obj.empty()
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py
class MyQueue:

    def __init__(self):
        self.stack_push = []
        self.stack_pop = []

    def push(self, x: int) -> None:
        self.stack_push.append(x)

    def pop(self) -> int:
        if not self.stack_pop:
            while self.stack_push:
                self.stack_pop.append(self.stack_push.pop())
        return self.stack_pop.pop()

    def peek(self) -> int:
        if not self.stack_pop:
            while self.stack_push:
                self.stack_pop.append(self.stack_push.pop())
        return self.stack_pop[-1]

    def empty(self) -> bool:
        return not self.stack_push and not self.stack_pop


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()